src/olympia/shelves/serializers.py

```diff
@@ -6,21 +6,66 @@
 
 from olympia import amo
 from olympia.addons.views import AddonSearchView
+from olympia.amo.reverse import reverse
+from olympia.amo.templatetags.jinja_helpers import absolutify
 from olympia.api.fields import (
     AbsoluteOutgoingURLField,
     GetTextTranslationSerializerField,
     ReverseChoiceField,
 )
+from olympia.api.utils import is_gate_active
 from olympia.bandwagon.views import CollectionAddonViewSet
-from olympia.hero.serializers import CTAField
 
 from .models import Shelf
 
 
-class ShelfFooterField(CTAField):
+class ShelfFooterField(serializers.Serializer):
     url = AbsoluteOutgoingURLField(source='footer_pathname')
     text = GetTextTranslationSerializerField(source='footer_text')
 
+    def to_representation(self, obj):
+        data = super().to_representation(obj)
+        request = self.context.get('request', None)
+        # when 'wrap-outgoing-parameter' is on url is a flat string already
+        is_flat_url = request and is_gate_active(request, 'wrap-outgoing-parameter')
+
+        url = data.get('url')
+        if not url and obj.endpoint in (
```

Comment:
Nit: I'm not sure if it would actually be better because of nesting, but looking at this it feels like you could start with just `if not url:`, because that is part of both of the following `if` / `elif` clauses.

Reply:
yeah I was just avoid the extra layer of indentation but I guess it's more "correct" to have a two levels.